src/CalcViewModel/GraphingCalculator/VariableViewModel.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "../Common/Utils.h"
+#include "EquationViewModel.h"
+
+namespace CalculatorApp::ViewModel
+{
+public
+    value struct VariableChangedEventArgs sealed
+    {
+        Platform::String ^ variableName;
+        double newValue;
+    };
+
+    [Windows::UI::Xaml::Data::Bindable] public ref class VariableViewModel sealed : public Windows::UI::Xaml::Data::INotifyPropertyChanged
+    {
+    public:
+        VariableViewModel(Platform::String ^ name, double value)
+            : m_Name(name)
+            , m_Value(value)
+            , m_SliderSettingsVisible(false)
+            , m_Min(0.0)
+            , m_Step(0.1)
+            , m_Max(2.0)
+        {
+        }
+
+        OBSERVABLE_OBJECT();
+        OBSERVABLE_PROPERTY_R(Platform::String ^, Name);
+        OBSERVABLE_PROPERTY_RW(double, Min);
```

Comment:
Since the work around is removed, we will need to pass the values through TrimTrailingZeroes in GraphingCalculator.xaml.cpp in the GraphingCalculator::SubmitTextbox function.

Reply:
Good catch!

I did a little different to manage all cases:
- 10.0000 -> 10
- 0000001 -> 1
- 1212. ->1212 
- '     12' -> 12
etc...

Instead of removing the trailing zeros, I convert the double back to string and replace the value.